packages/cxl-ui/src/components/cxl-jw-player/mixins/TranscriptMixin.js

```diff
@@ -0,0 +1,191 @@
+import { property, state, query } from 'lit/decorators.js';
+import { debounce } from 'lodash-es';
+import Mark from 'mark.js';
+import { parseSync } from 'subtitle';
+
+export function TranscriptMixin(BaseClass) {
+  class Mixin extends BaseClass {
+    _debouncedSearch;
+
+    _mark;
+
+    @property({ reflect: true, type: Boolean }) captions = false;
+    @property({ attribute: 'has-captions', reflect: true, type: Boolean }) hasCaptions = false;
```

Comment:
Captions can be enabled in video settings:
![Screenshot 2024-09-20 at 14 10 32](https://github.com/user-attachments/assets/733787d2-8bd1-49e1-949f-320aee1bff87)

but we're in the "transcript" class so, shouldn't it be named "has-transcript"?

Reply:
Captions and transcript are pretty much the same thing. There was a requirement for the full text to be available to read. I prefer naming everything captions right now, I believe Syed renamed a few things and didn't finish it.